packages/fhir-router/src/batch.ts

```diff
@@ -145,16 +145,13 @@ class BatchProcessor {
       body = this.parsePatchBody(entry);
     }
 
-    // Pass in dummy host for parsing purposes.
-    // The host is ignored.
-    const url = new URL(request.url as string, 'https://example.com/');
-
     const result = await this.router.handleRequest(
       {
         method: request.method as HttpMethod,
-        pathname: url.pathname,
+        pathname: request.url,
+        // Params and query string are parsed from the `pathname` by the router
         params: Object.create(null),
-        query: Object.fromEntries(url.searchParams),
+        query: Object.create(null),
```

Comment:
I see what you're going for here.  I feel like this is now more loose and ambiguous than before though 😞 

How nasty would it be to remove `pathname` and `query`, and replace them with `url` for consistency?

Or is the idea that `FhirRequest` is now a 2-phase object?  There's the input value, where `pathname === url`, and then there's the pre-processed value that gets passed to the handlers, where `query` is populated?